### DISCUSSION TITLE
Channels | Learn Go

### DISCUSSION THREAD

**Discussion by giscus** [karanpratapsingh/portfolio#42] [Comments]
# Channels | Learn Go

In this tutorial, we will learn about channels in Go

https://www.karanpratapsingh.com/courses/go/channels

**Comment by dineshtbits** [comment]
Hi Karan, 

Thanks for writing this course, it is super helpful. 

A quick question on this topic. 

The following code snippet works,

```go
func main() {
	ch := make(chan string)
	go func() {
		ch <- "Hello"
	}()
	fmt.Println("message", <-ch)
}
```

However, this doesn't 
```go
func main() {
	ch := make(chan string)
	ch <- "Hello"
	fmt.Println("message", <-ch)
}
```

The only difference is sending a message via a new go routine vs. the same go routine (main). Just curious.
